File: transportService/src/main/java/com/easytrip/userservice/models/TransportWithUserDTO.java
package com.easytrip.userservice.models;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class TransportWithUserDTO {
    private String id;
    private String voyageId;
    private String type;
    private String compagnie;
    private int capacite;
    private String numero;
    private String villeDepart;
    private String villeArrivee;
    private LocalDateTime dateDepart;
    private LocalDateTime dateArrivee;
    private double prix;

    private UserDTO user;
}

File: FrontEnd/user-serviceFront/src/app/models/TransportWithUserDTO.ts
// transport-with-user-dto.ts
export interface TransportWithUserDTO {
    id: string;
    voyageId: string;
    type: string;
    compagnie: string;
    capacite: number;
    numero: string;
    villeDepart: string;
    villeArrivee: string;
    dateDepart: string;
    dateArrivee: string;
    prix: number;
    user: {
      id: string;
      name: string;
      email: string;
      phone: string;
    };
  }
  